server/server.go

```diff
@@ -1073,12 +1076,37 @@ func (s *Server) handleBodyAsMessageAutoDetect(m *message, body *util.PeekedRead
 	return nil
 }
 
-func (s *Server) handleBodyAsTextMessage(m *message, body *util.PeekedReadCloser) error {
+func (s *Server) handleBodyAsTextMessage(m *message, body *util.PeekedReadCloser, template bool) error {
 	if !utf8.Valid(body.PeekedBytes) {
 		return errHTTPBadRequestMessageNotUTF8.With(m)
 	}
 	if len(body.PeekedBytes) > 0 { // Empty body should not override message (publish via GET!)
-		m.Message = strings.TrimSpace(string(body.PeekedBytes)) // Truncates the message to the peek limit if required
+		peakedBody := strings.TrimSpace(string(body.PeekedBytes)) // Truncates the message to the peek limit if required
+		if template && gjson.Valid(peakedBody) {
+			// Replace JSON paths in message
+			r := regexp.MustCompile(`\${([^}]+)}`)
+			matches := r.FindAllStringSubmatch(m.Message, -1)
+			for _, v := range matches {
+				query := v[1]
+				result := gjson.Get(peakedBody, query)
+				if result.Exists() {
+					m.Message = strings.ReplaceAll(m.Message, fmt.Sprintf("${%s}", query), result.String())
```

Comment:
What happens if `result` is a number or an array?

Reply:
This works fine. See TestServer_MessageTemplate_FancyGJSON (server/server_test.go:L2742)